jaeger.yaml

```diff
@@ -0,0 +1,71 @@
+package:
+  name: jaeger
+  version: 1.61.0
+  epoch: 0
+  description: Jaeger, a Distributed Tracing Platform
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/jaegertracing/jaeger
+      tag: v${{package.version}}
+      expected-commit: e047f2091158d7c9b3e9e12afa9f74de0da02f43
+      recurse-submodules: true
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - nodejs-20
+      - yarn
+
+data:
+  - name: jaeger-components
+    items:
+      agent: "."
+      all-in-one: "."
+      anonymizer: "."
+      collector: "."
+      es-index-cleaner: "."
+      es-rollover: "."
+      esmapping-generator: "."
+      ingester: "."
+      query: "."
+      remote-storage: "."
+      tracegen: "."
+
+subpackages:
+  - range: jaeger-components
+    name: "${{package.name}}-${{range.key}}"
+    pipeline:
+      - runs: |
+          ## Only run this step if building for all-in-one package
+          ## This builds the ui component of jaeger with yarn
+          if [[ "${{range.key}}" = "all-in-one" ]]; then
+            mkdir -p jaeger-ui/packages/jaeger-ui/build
+            rm -rf cmd/query/app/ui/actual/*
+            yarn install --frozen-lockfile --cwd jaeger-ui
+            cd jaeger-ui/packages/jaeger-ui/build
+            yarn build
+          fi
+      - runs: |
+          ## Only run this step if building for all-in-one package
+          ## This copies the build artifact into a go package
+          if [[ "${{range.key}}" = "all-in-one" ]]; then
+            cp -r jaeger-ui/packages/jaeger-ui/build/* cmd/query/app/ui/actual
+            find cmd/query/app/ui/actual -type f | grep -v .gitignore | xargs gzip --no-name
+            touch -t $(date -r jaeger-ui/packages/jaeger-ui/build/index.html '+%Y%m%d%H%M.%S') cmd/query/app/ui/actual/index.html.gz
+          fi
+      - uses: go/build
+        with:
+          modroot: cmd/${{range.key}}
+          packages: .
+          ldflags: -X github.com/jaegertracing/jaeger/pkg/version.latestVersion=v${{package.version}}
+          output: ${{range.key}}
+
```

Comment:
Can we add simple tests? i.e. `--help` or `version`.

Preferably `test/daemon-check-output`.

Reply:
Have added tests that encompass both version and help for those packages that have them.

Reply:
Nice, thanks! You can create the tests under subpackages-level so you don't need to write every subpackage under the test block. See this, for example: https://github.com/wolfi-dev/os/blob/4bdb602db133a030a33f39f946d893c686530d52/nvidia-container-toolkit.yaml#L57-L61

In this case, you can move the `version` and `help` checks to subpackage block:

```
test:
      pipeline:
        - runs: |
            ${{range.key}} version 2>&1 | grep ${{package.version}}
            ${{range.key}} --help
```

Reply:
The version test is not available for all the components. Moving this into the subpackage block will break the tests as not all the components have the version command. 

A conditional statement on the version test will be unwieldly

Reply:
This looks fine to me